src/elfpy/simulators.py

```diff
@@ -300,6 +315,16 @@ def run_simulation(self, override_dict=None):
 
     def collect_and_execute_trades(self, last_block_in_sim=False):
         """Get trades from the agent list, execute them, and update states"""
+        if not isinstance(self.market, Market):
+            raise ValueError("market not defined")
+
+        # TODO: BIG HACK to make this PR pass.  There is a bug with removing liquidity that
+        # doesn't account for open shorts.  Someone can remove all liquidity while there are open
+        # shorts, then when another user goes to close a short, there are no reserves, so the calc
+        # functions error due to division by zero.  Need to fix the accounting of reserves to
+        # account for open short positions.
+        if last_block_in_sim:  # get all of a agent's trades
```

Comment:
maybe change this comment to "reverse trade order so that init_lp is last" to remind us what `[::-1]` does.